src/app/common/ResourceCodeBlock.vue

```diff
@@ -0,0 +1,141 @@
+          @click="copyAsKubernetes"
+        >
+          <KIcon
+            color="currentColor"
+            icon="copy"
+            size="18"
+            :title="i18n.t('common.copyKubernetesText')"
+          />
+
+          {{ i18n.t('common.copyKubernetesShortText') }}
+        </KButton>
+      </template>
+    </CodeBlock>
+  </StatusInfo>
+</template>
+
+<script lang="ts" setup>
+import { KButton, KIcon } from '@kong/kongponents'
+import { PropType, ref, watch } from 'vue'
+
+import CodeBlock from './CodeBlock.vue'
+import StatusInfo from './StatusInfo.vue'
+import type { SingleResourceParameters } from '@/types/api.d'
+import type { Entity } from '@/types/index.d'
+import { useI18n } from '@/utilities'
+import { copyTextToClipboard } from '@/utilities/copyTextToClipboard'
+import { toYaml } from '@/utilities/toYaml'
+
+const i18n = useI18n()
+
+const props = defineProps({
+  id: {
+    type: String,
+    required: true,
+  },
+
+  /**
+   * Function returning the resource.
+   */
+  resourceFetcher: {
+    type: Function as PropType<(params?: SingleResourceParameters) => Promise<Entity>>,
+    required: true,
+  },
+
+  /**
+   * Triggers re-fetching the resource when set to a new value. Provide it with an updating resource name, for example, to update rendered content.
+   */
+  resourceFetcherWatchKey: {
+    type: [String, Number] as PropType<string | number | null>,
+    required: false,
+    default: null,
+  },
+
+  codeMaxHeight: {
+    type: String,
+    required: false,
+    default: null,
+  },
+
+  isSearchable: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+})
+
+const yamlUniversal = ref('')
+const isLoading = ref(false)
+const error = ref<Error | null>(null)
+
+watch(() => props.resourceFetcherWatchKey, function (newKey, oldKey) {
+  if (newKey !== oldKey && newKey !== null) {
+    fetchResource()
+  }
+})
+
+watch(() => props.resourceFetcher, function () {
+  fetchResource()
+}, { immediate: true })
+
+async function fetchResource() {
+  isLoading.value = true
+  error.value = null
+
+  try {
+    const resourceUniversal = await props.resourceFetcher()
+
+    yamlUniversal.value = toYamlRepresentation(resourceUniversal)
+  } catch (err) {
+    if (err instanceof Error) {
+      error.value = err
+    } else {
+      console.error(err)
+    }
+  } finally {
+    isLoading.value = false
+  }
+}
+
+async function copyAsKubernetes() {
+  error.value = null
+
+  try {
+    const resourceKubernetes = await props.resourceFetcher({ format: 'kubernetes' })
+
+    const yamlKubernetes = toYamlRepresentation(resourceKubernetes)
+    await copyTextToClipboard(yamlKubernetes)
+  } catch (err) {
+    if (err instanceof Error) {
+      error.value = err
+    } else {
+      console.error(err)
+    }
+  }
+}
+
+function toYamlRepresentation(resource: Entity): string {
+  const { creationTime, modificationTime, ...resourceWithoutTimes } = resource
+
+  return toYaml(resourceWithoutTimes)
+}
+</script>
```

Comment:
Personally I don't think we should be embedding data retrieval/loading/error support in a presentation component. I can kinda understand how you're thinking, but we should work on moving towards only passing data down into presentational components, thus decoupling them from how they end up with their data. Not a blocker right now tho.